src/mesh/impls/bout/boutmesh.cxx

```diff
@@ -127,8 +127,20 @@ int BoutMesh::load() {
   output << "\tGrid size: " << nx << " x " << ny << " x " << MZ << endl;
 
   // Get guard cell sizes
-  options->get("MXG", MXG, 2);
-  options->get("MYG", MYG, 2);
+  // Try to read from grid file first, then if not found
+  // get from options
+  if (Mesh::get(MXG, "MXG")) {
+    // Error code returned
+    options->get("MXG", MXG, 2);
+  }
+  ASSERT0(MXG >= 0);
```

Comment:
ASSERT0 is always checked, since if CHECK is not given the level is set to zero:
https://github.com/boutproject/BOUT-dev/blob/master/include/bout/assert.hxx#L29